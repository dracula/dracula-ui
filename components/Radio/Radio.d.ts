import React, { HTMLAttributes } from 'react';
import { ColorMap } from '../../base/colors';
import { MarginMixin, PaddingMixin } from '../../base/spacing';
export declare const radioColors: Partial<ColorMap>;
/** Radio Props */
export interface RadioProps extends HTMLAttributes<HTMLInputElement>, PaddingMixin, MarginMixin {
    /** The Dracula UI theme color for the radio button. */
    color: keyof typeof radioColors;
    /** The name of the radio. Mirrors the name HTML attribute. */
    name?: string;
    /** A disabled radio is unusable and un-clickable. */
    disabled?: boolean;
}
/**
 * Radio is a styled HTML Input of type radio.
 *
 * There are no behavior changes applied to the native HTML tag other
 * than light styling done via CSS in order to keep Radios accessible.
 */
export declare const Radio: React.FC<RadioProps>;
