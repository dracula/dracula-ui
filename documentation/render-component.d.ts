import React from 'react';
export declare type ComponentExample = {
    html: string;
    docs: string;
    title: string;
    react?: any;
    screenshot?: string;
    svgData?: string;
};
export declare type SnapshotBuilder = {
    comp: () => React.ReactElement;
    docs: string;
    title: string;
};
export declare function snapshot(title: string, comp: () => React.ReactElement, docs?: string): SnapshotBuilder;
export declare function renderSnapshot(title: string, comp: () => React.ReactElement, docs?: string): ComponentExample;
