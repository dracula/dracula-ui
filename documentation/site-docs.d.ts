import { SnapshotBuilder } from './render-component';
import React from 'react';
interface Documentation {
    basic: () => SnapshotBuilder;
    variations: () => Array<SnapshotBuilder> | SnapshotBuilder;
}
export declare function docs<T>(componentClass: React.FunctionComponent<T> | string, documentation: Documentation): void;
export {};
