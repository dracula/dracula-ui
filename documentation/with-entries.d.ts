import { SnapshotBuilder } from './render-component';
export declare function withEntries<T>(map: T, fn: (name: keyof T, value: T[keyof T]) => SnapshotBuilder): SnapshotBuilder[];
