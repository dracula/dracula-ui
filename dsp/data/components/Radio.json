{
  "class": "component",
  "type": "page",
  "id": "Radio",
  "name": "Radio",
  "last_updated": "2021-03-14T22:34:40.925Z",
  "last_updated_by": "System",
  "description": "Radio is a styled HTML Input of type radio.\n\nThere are no behavior changes applied to the native HTML tag other\nthan light styling done via CSS in order to keep Radios accessible.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "radio",
    "languages": {
      "javascriptreact": "<div>\n  <Radio id=\"one\" defaultChecked={true} color=\"purple\" name=\"demo\" />\n  <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n    Checked\n  </label>\n</div>;\n",
      "html": "<div>\n  <input\n    type=\"radio\"\n    id=\"one\"\n    color=\"purple\"\n    name=\"demo\"\n    class=\"drac-radio drac-radio-purple\"\n    checked=\"\"\n  /><label for=\"one\" class=\"drac-text drac-text-white\">Checked</label>\n</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div>\n  <input\n    type=\"radio\"\n    id=\"one\"\n    color=\"purple\"\n    name=\"demo\"\n    class=\"drac-radio drac-radio-purple\"\n    checked=\"\"\n  /><label for=\"one\" class=\"drac-text drac-text-white\">Checked</label>\n</div>\n",
      "react": "<div>\n  <Radio id=\"one\" defaultChecked={true} color=\"purple\" name=\"demo\" />\n  <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n    Checked\n  </label>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Radio.png",
      "svgData": "./dsp/assets/svgs/Radio.svg"
    },
    {
      "title": "Disabled",
      "html": "<div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"one\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-white\"\n      checked=\"\"\n    /><label for=\"one\" class=\"drac-text drac-text-white\">Checked</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"two\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-white\"\n    /><label for=\"two\" class=\"drac-text drac-text-white\">Unchecked</label>\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"one\" defaultChecked={true} color=\"white\" name=\"demo\" />\n    <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n      Checked\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"two\" color=\"white\" name=\"demo\" />\n    <label htmlFor=\"two\" className=\"drac-text drac-text-white\">\n      Unchecked\n    </label>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/RadioDisabled.png",
      "svgData": "./dsp/assets/svgs/RadioDisabled.svg"
    },
    {
      "title": "Colors",
      "html": "<div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"white\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-white\"\n    /><label for=\"white\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"cyan\"\n      color=\"cyan\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-cyan\"\n    /><label for=\"cyan\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"green\"\n      color=\"green\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-green\"\n    /><label for=\"green\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"orange\"\n      color=\"orange\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-orange\"\n    /><label for=\"orange\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"pink\"\n      color=\"pink\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-pink\"\n    /><label for=\"pink\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"purple\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-purple\"\n    /><label for=\"purple\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"red\"\n      color=\"red\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-red\"\n    /><label for=\"red\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"yellow\"\n      color=\"yellow\"\n      name=\"demo\"\n      class=\"drac-radio drac-radio-yellow\"\n    /><label for=\"yellow\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"animated\"\n      color=\"animated\"\n      name=\"demo\"\n      class=\"drac-radio\"\n    /><label for=\"animated\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"cyanGreen\"\n      color=\"cyanGreen\"\n      name=\"demo\"\n      class=\"drac-radio\"\n    /><label for=\"cyanGreen\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"pinkPurple\"\n      color=\"pinkPurple\"\n      name=\"demo\"\n      class=\"drac-radio\"\n    /><label for=\"pinkPurple\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"purpleCyan\"\n      color=\"purpleCyan\"\n      name=\"demo\"\n      class=\"drac-radio\"\n    /><label for=\"purpleCyan\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"radio\"\n      id=\"yellowPink\"\n      color=\"yellowPink\"\n      name=\"demo\"\n      class=\"drac-radio\"\n    /><label for=\"yellowPink\" class=\"drac-text drac-text-white\">Radio</label>\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"white\" color=\"white\" name=\"demo\" />\n    <label htmlFor=\"white\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"cyan\" color=\"cyan\" name=\"demo\" />\n    <label htmlFor=\"cyan\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"green\" color=\"green\" name=\"demo\" />\n    <label htmlFor=\"green\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"orange\" color=\"orange\" name=\"demo\" />\n    <label htmlFor=\"orange\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"pink\" color=\"pink\" name=\"demo\" />\n    <label htmlFor=\"pink\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"purple\" color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"purple\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"red\" color=\"red\" name=\"demo\" />\n    <label htmlFor=\"red\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"yellow\" color=\"yellow\" name=\"demo\" />\n    <label htmlFor=\"yellow\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"animated\" color=\"animated\" name=\"demo\" />\n    <label htmlFor=\"animated\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"cyanGreen\" color=\"cyanGreen\" name=\"demo\" />\n    <label htmlFor=\"cyanGreen\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"pinkPurple\" color=\"pinkPurple\" name=\"demo\" />\n    <label htmlFor=\"pinkPurple\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"purpleCyan\" color=\"purpleCyan\" name=\"demo\" />\n    <label htmlFor=\"purpleCyan\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Radio id=\"yellowPink\" color=\"yellowPink\" name=\"demo\" />\n    <label htmlFor=\"yellowPink\" className=\"drac-text drac-text-white\">\n      Radio\n    </label>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/RadioColors.png",
      "svgData": "./dsp/assets/svgs/RadioColors.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The Dracula UI theme color for the radio button.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Radio/Radio.tsx",
        "name": "RadioProps"
      },
      "declarations": [
        { "fileName": "src/components/Radio/Radio.tsx", "name": "RadioProps" }
      ],
      "required": true,
      "type": {
        "name": "\"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\""
      }
    },
    "name": {
      "defaultValue": null,
      "description": "The name of the radio. Mirrors the name HTML attribute.",
      "name": "name",
      "parent": {
        "fileName": "src/components/Radio/Radio.tsx",
        "name": "RadioProps"
      },
      "declarations": [
        { "fileName": "src/components/Radio/Radio.tsx", "name": "RadioProps" }
      ],
      "required": false,
      "type": { "name": "string | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Radio is a styled HTML Input of type radio.\n\nThere are no behavior changes applied to the native HTML tag other\nthan light styling done via CSS in order to keep Radios accessible.",
    "displayName": "Radio",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The Dracula UI theme color for the radio button.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Radio/Radio.tsx",
          "name": "RadioProps"
        },
        "declarations": [
          { "fileName": "src/components/Radio/Radio.tsx", "name": "RadioProps" }
        ],
        "required": true,
        "type": {
          "name": "\"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\""
        }
      },
      "name": {
        "defaultValue": null,
        "description": "The name of the radio. Mirrors the name HTML attribute.",
        "name": "name",
        "parent": {
          "fileName": "src/components/Radio/Radio.tsx",
          "name": "RadioProps"
        },
        "declarations": [
          { "fileName": "src/components/Radio/Radio.tsx", "name": "RadioProps" }
        ],
        "required": false,
        "type": { "name": "string | undefined" }
      }
    }
  }
}
