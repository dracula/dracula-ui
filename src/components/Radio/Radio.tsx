import { ColorMap } from '../../base/colors'
import React, { HTMLAttributes } from 'react'

export const radioColors: ColorMap = {
  white: 'drac-radio-white',
  cyan: 'drac-radio-cyan',
  green: 'drac-radio-green',
  orange: 'drac-radio-orange',
  pink: 'drac-radio-pink',
  purple: 'drac-radio-purple',
  red: 'drac-radio-red',
  yellow: 'drac-radio-yellow',
  animated: '',
  cyanGreen: '',
  pinkPurple: '',
  purpleCyan: '',
  yellowPink: ''
}

/** Radio Props */
export interface RadioProps extends HTMLAttributes<HTMLInputElement> {
  /** The Dracula UI theme color for the radio button. */
  color: keyof typeof radioColors

  /** The name of the radio. Mirrors the name HTML attribute. */
  name?: string
}

/**
 * Radio is a styled HTML Input of type radio.
 *
 * There are no behavior changes applied to the native HTML tag other
 * than light styling done via CSS in order to keep Radios accessible.
 */
export const Radio: React.FC<RadioProps> = (props: RadioProps) => {
  const finalProps = {
    ...props,
    className: `drac-radio ${radioColors[props.color]}`
  }

  return <input type="radio" {...finalProps} />
}

Radio.displayName = 'Radio'
