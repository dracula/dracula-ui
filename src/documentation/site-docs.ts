import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '../story-helpers/render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from '../story-helpers/pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

jest.setTimeout(10000)

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name
  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeEach(async () => {
      browser = await puppeteer.launch({ headless: true, devtools: true })
    })

    afterEach(async () => {
      await browser.close()
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      const [screenshot, svg] = await componentScreenshot(
        browser,
        snapshot,
        null,
        name
      )

      snapshot.screenshot = screenshot
      snapshot.svgData = svg

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          variation,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
