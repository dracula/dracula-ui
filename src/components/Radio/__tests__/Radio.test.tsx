import { Radio } from '@/components/Radio/Radio'
import {
  buildSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import { siteDocs } from '@/documentation/site-docs'
import {
  RadioNormal,
  RadioDisabled,
  RadioColors
} from '@/components/Radio/__stories__/Radio.stories'

siteDocs(Radio, {
  basicUsage() {
    return buildSnapshot('Usage', RadioNormal)
  },
  variations() {
    return [
      buildSnapshot('Disabled', RadioDisabled),
      buildSnapshot('Colors', RadioColors)
    ] as SnapshotBuilder[]
  }
})
