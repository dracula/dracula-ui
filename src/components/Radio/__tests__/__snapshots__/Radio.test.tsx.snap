// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Radio Basic Usage 1`] = `
Object {
  "docs": "",
  "html": "<div>
  <input
    type=\\"radio\\"
    id=\\"one\\"
    color=\\"purple\\"
    name=\\"demo\\"
    class=\\"drac-radio drac-radio-purple\\"
    checked=\\"\\"
  /><label for=\\"one\\" class=\\"drac-text drac-text-white\\">Checked</label>
</div>
",
  "react": "<div>
  <Radio id=\\"one\\" defaultChecked={true} color=\\"purple\\" name=\\"demo\\" />
  <label htmlFor=\\"one\\" className=\\"drac-text drac-text-white\\">
    Checked
  </label>
</div>;
",
  "title": "Usage",
}
`;

exports[`Site: Radio Colors 1`] = `
Object {
  "docs": "",
  "html": "<div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"white\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-white\\"
    /><label for=\\"white\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"cyan\\"
      color=\\"cyan\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-cyan\\"
    /><label for=\\"cyan\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"green\\"
      color=\\"green\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-green\\"
    /><label for=\\"green\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"orange\\"
      color=\\"orange\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-orange\\"
    /><label for=\\"orange\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"pink\\"
      color=\\"pink\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-pink\\"
    /><label for=\\"pink\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"purple\\"
      color=\\"purple\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-purple\\"
    /><label for=\\"purple\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"red\\"
      color=\\"red\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-red\\"
    /><label for=\\"red\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"yellow\\"
      color=\\"yellow\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-yellow\\"
    /><label for=\\"yellow\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"animated\\"
      color=\\"animated\\"
      name=\\"demo\\"
      class=\\"drac-radio\\"
    /><label for=\\"animated\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"cyanGreen\\"
      color=\\"cyanGreen\\"
      name=\\"demo\\"
      class=\\"drac-radio\\"
    /><label for=\\"cyanGreen\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"pinkPurple\\"
      color=\\"pinkPurple\\"
      name=\\"demo\\"
      class=\\"drac-radio\\"
    /><label for=\\"pinkPurple\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"purpleCyan\\"
      color=\\"purpleCyan\\"
      name=\\"demo\\"
      class=\\"drac-radio\\"
    /><label for=\\"purpleCyan\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"yellowPink\\"
      color=\\"yellowPink\\"
      name=\\"demo\\"
      class=\\"drac-radio\\"
    /><label for=\\"yellowPink\\" class=\\"drac-text drac-text-white\\">Radio</label>
  </div>
</div>
",
  "react": "<div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"white\\" color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"white\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"cyan\\" color=\\"cyan\\" name=\\"demo\\" />
    <label htmlFor=\\"cyan\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"green\\" color=\\"green\\" name=\\"demo\\" />
    <label htmlFor=\\"green\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"orange\\" color=\\"orange\\" name=\\"demo\\" />
    <label htmlFor=\\"orange\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"pink\\" color=\\"pink\\" name=\\"demo\\" />
    <label htmlFor=\\"pink\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"purple\\" color=\\"purple\\" name=\\"demo\\" />
    <label htmlFor=\\"purple\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"red\\" color=\\"red\\" name=\\"demo\\" />
    <label htmlFor=\\"red\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"yellow\\" color=\\"yellow\\" name=\\"demo\\" />
    <label htmlFor=\\"yellow\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"animated\\" color=\\"animated\\" name=\\"demo\\" />
    <label htmlFor=\\"animated\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"cyanGreen\\" color=\\"cyanGreen\\" name=\\"demo\\" />
    <label htmlFor=\\"cyanGreen\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"pinkPurple\\" color=\\"pinkPurple\\" name=\\"demo\\" />
    <label htmlFor=\\"pinkPurple\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"purpleCyan\\" color=\\"purpleCyan\\" name=\\"demo\\" />
    <label htmlFor=\\"purpleCyan\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"yellowPink\\" color=\\"yellowPink\\" name=\\"demo\\" />
    <label htmlFor=\\"yellowPink\\" className=\\"drac-text drac-text-white\\">
      Radio
    </label>
  </div>
</div>;
",
  "title": "Colors",
}
`;

exports[`Site: Radio Disabled 1`] = `
Object {
  "docs": "",
  "html": "<div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"one\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-white\\"
      checked=\\"\\"
    /><label for=\\"one\\" class=\\"drac-text drac-text-white\\">Checked</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"radio\\"
      id=\\"two\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-radio drac-radio-white\\"
    /><label for=\\"two\\" class=\\"drac-text drac-text-white\\">Unchecked</label>
  </div>
</div>
",
  "react": "<div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"one\\" defaultChecked={true} color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"one\\" className=\\"drac-text drac-text-white\\">
      Checked
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Radio id=\\"two\\" color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"two\\" className=\\"drac-text drac-text-white\\">
      Unchecked
    </label>
  </div>
</div>;
",
  "title": "Disabled",
}
`;
